src/vs/workbench/contrib/chat/browser/chatSetup.ts

```diff
@@ -1089,6 +1106,101 @@ class ChatSetupWelcomeContent extends Disposable {
 		button.label = buttonLabel;
 		button.enabled = this.controller.step === ChatSetupStep.Initial;
 	}
+
+	private async updateProvider(useGhe: boolean): Promise<void> {
+		const registry = Registry.as<IConfigurationRegistry>(ConfigurationExtensions.Configuration);
+		registry.registerConfiguration({
+			'id': 'copilot.setup',
+			'type': 'object',
+			'properties': {
+				[defaultChat.providerSetting]: {
+					'type': 'string'
+				},
+				[defaultChat.providerUriSetting]: {
+					'type': 'string'
+				}
+			}
+		});
```

Comment:
whoa, you register a configuration? Is an extension allowed to do this too?

Reply:
In my testing, I did not encounter any issues when the configuration was registered in core and then later provided by the extension or vice versa, but maybe @sandy081 or @aeschli can chime in if registering the same setting multiple times is an issue.